src/compiler/program.ts

```diff
@@ -2031,6 +2031,7 @@ namespace ts {
             redirect.resolvedPath = resolvedPath;
             redirect.originalFileName = originalFileName;
             redirect.redirectInfo = { redirectTarget, unredirected };
+            sourceFilesFoundSearchingNodeModules.set(path, !!sourceFilesFoundSearchingNodeModules.get(redirectTarget.path));

```

Comment:
No. This doesn't work either. You want to set it just like normal sourceFile that is with `sourceFilesFoundSearchingNodeModules.set(path, currentNodeModulesDepth > 0);` otherwise you need to handle many more cases when we find sourceFile and aren't looking for currentNodeModules and set it to false. (from code 
```typescript
if (file && sourceFilesFoundSearchingNodeModules.get(file.path) && currentNodeModulesDepth === 0) {
                    sourceFilesFoundSearchingNodeModules.set(file.path, false);
```